mdata/aggmetric.go

```diff
@@ -163,25 +163,38 @@ func (a *AggMetric) GetAggregated(consolidator consolidation.Consolidator, aggSp
 				badConsolidator.Inc()
 				return Result{}, err
 			case consolidation.Cnt:
-				agg = a.cntMetric
+				agg = aggregator.cntMetric
 			case consolidation.Lst:
-				agg = a.lstMetric
+				agg = aggregator.lstMetric
 			case consolidation.Min:
-				agg = a.minMetric
+				agg = aggregator.minMetric
 			case consolidation.Max:
-				agg = a.maxMetric
+				agg = aggregator.maxMetric
 			case consolidation.Sum:
-				agg = a.sumMetric
+				agg = aggregator.sumMetric
 			default:
 				err := fmt.Errorf("internal error: AggMetric.GetAggregated(): unknown consolidator %q", consolidator)
 				log.Errorf("AM: %s", err.Error())
 				badConsolidator.Inc()
 				return Result{}, err
 			}
 			if agg == nil {
-				return Result{}, fmt.Errorf("Consolidator %q not configured", consolidator)
+				return Result{}, fmt.Errorf("consolidator %q not configured", consolidator)
+			}
+			result, err := agg.Get(from, to)
+			if err != nil {
+				return Result{}, err
 			}
-			return agg.Get(from, to)
+			futurePoints := []schema.Point{}
+			if a.rob != nil {
+				futurePoints = a.rob.Get()
+			}
+			result.Points = aggregator.Foresee(consolidator, from, to, futurePoints)
+			if result.Oldest > result.Points[0].Ts {
+				result.Oldest = result.Points[0].Ts
+			}
```

Comment:
Couldn't all of this be in the `if` statement above? That reduces the changes required for the non-rob case.